syntax = "proto3";

package TaskListPackage;

service TaskList {
    rpc CreateTask (CreateTaskMessage) returns (TaskCreatedMessage);
    rpc ListTasks (Empty) returns (TaskListMessage);
    rpc CompleteTask (CompleteTaskMessage) returns (Empty);
    rpc CalculateRiskScore (RiskScoreRequest) returns (RiskScoreUpdate);
}

message Empty {}

message CreateTaskMessage {
    string title = 1;
    string description = 2;
}

message TaskCreatedMessage {
    uint32 id = 1;
}

message TaskMessage {
    uint32 id = 1;
    string title = 2;
    string description = 3;
    bool completed = 4;
}

message TaskListMessage {
    repeated TaskMessage tasks = 1;
}

message CompleteTaskMessage {
    uint32 id = 1;
}

message RiskScoreRequest {
    string manufacturer = 1;
    string device = 2;
}
message RiskScoreUpdate {
    string status = 1;
    double progress = 2;
    double risk_score = 3;
    int32 record_count = 4;
    string message = 5;
    bool completed = 6;
}